### ISSUE TITLE
Error in coastline extraction

### ISSUE THREAD

**Issue by EmaueleIngrassia** [kvos/CoastSat#154]
Hello =) I like a lot CoastSat, thanks for everything. I'm working in a coastal management area close to a river mouth. Coordinates are : polygon = [[[19.0811811798211,  40.84662584964178],
            [19.64642059745296, 40.84662584964178],
            [19.64642059745296, 40.25081828816741 ],
            [19.0811811798211, 40.25081828816741 ],
            [19.0811811798211,  40.84662584964178]]]
The main problem is that probably there is too many river debris close to the coast, and for this CoastSat don't recognize the differences between beach and sea. The output that I have is an incomplete coastline that is quite impossible to fix also on Qgis.
Should I modify some part of code to change the parameter with CoastSat attributes values  to the pixels? Do you think that this could be a good option? or maybe the software will recognize the coastline away from the real coast?
Thanks a lot!!
![Vjosa](https://user-images.githubusercontent.com/74313916/98834894-04aebf00-2440-11eb-93fd-e8edf910b336.PNG)

**Comment by kvos** [comment]
did you define a reference shoreline? try increasing the parameter called max_dist_ref as it seems to me that those shoreline points are too far from your reference shoreline.

**Comment by EmaueleIngrassia** [comment]
Yes, I defined a shoreline as well done as possible, but unfortunately, I still have this problem. I would also like to underline that this images has less debris in the sea then previous. 
I fixed max_dist_ref to 200m. 
This output is better the previous, but still is still difficult to work with this output on Qgis to create a line that shows the coastline

![Vjosa 2 shoreline output](https://user-images.githubusercontent.com/74313916/99150587-f6f27700-2695-11eb-8d36-95e03df3fe82.JPG)
![Vjosa 2 shoreline](https://user-images.githubusercontent.com/74313916/99150588-f823a400-2695-11eb-8a23-6601df763f1d.JPG)

**Comment by kvos** [comment]
I don't think debris are an issue. try increasing max dist ref to 400 and decreasing min_sl_length to 200

**Comment by EmaueleIngrassia** [comment]
Ehy, sorry but I tried to fix everything before report here the results. 
I managed the data and first of all I reduced the ROI. It was too big. Then I realised that CoastSat was showing the coastline in a buffer area that was too big, in fact, it was detecting also the coastline of some coastal lakes, and it was the biggest problem. I reduced the buffer area to 100m from the reference shoreline and I set the minimum length of a coastline to 100m to exclude shoreline too short. 

Honestly, the possibility in the Master version to manage the threshold helped a lot to improve the detection.
![FOCEVJOSAGithub](https://user-images.githubusercontent.com/74313916/101985335-ac115280-3c87-11eb-9888-01c2c61d6331.png)

Anyway, this version continues to give some problems, in particular in the mine ROIs, not only that one showed in the photo, CoastSat continued to raise an exception. 
![immagine](https://user-images.githubusercontent.com/74313916/101985446-5ab59300-3c88-11eb-9a9b-e6104dec9e44.png)

Seems like the gds_all are not declared. This happens only to me? or someone has the same problem? 
I tried changing ROI and dates, but a lot of times it doesn't works
